### DISCUSSION TITLE
使用 JSON Schema 描述 next-sheet 中的数据结构

### DISCUSSION THREAD

**Discussion by AlanLang** [next-sheet/next-sheet#5] [Ideas]
# 使用 JSON Schema 描述 next-sheet 中的数据结构
JSON Schema 是一种描述数据结构的通用格式，我们用 JSON Schema 来描述 next-sheet 中的数据结构，
这样就可以使用 JSON Schema 的工具来验证数据的合法性，也可以使用工具来生成数据的模板。

## 相关工具
* [生成数据模板](https://app.quicktype.io/)
* [可视化编辑](https://hellosean1025.github.io/json-schema-visual-editor/)
* [在线转换](https://transform.tools/json-schema-to-typescript)

**Comment by nwyle** [comment]
json schema怎么表达，当type = 'image'时，payload结构为xxx；当type = 'text'是，payload结构为yyy，这种需求

**Reply by AlanLang** [reply]
JSON Schema 是可以写条件语句的：https://json-schema.apifox.cn/#有条件地应用子模式
